qp_klp/Amplicon.py

```diff
@@ -81,23 +85,57 @@ def quality_control(self):
                 new_path = join(output_folder, basename(raw_fastq_file))
                 copyfile(raw_fastq_file, new_path)
 
-    def generate_reports(self, input_file_path):
-        super()._generate_reports(self.pipeline.mapping_file)
+    def generate_reports(self):
+        super()._generate_reports()
         return None  # amplicon doesn't need project names
 
+    def _get_data_type(self, prep_file_path):
+        metadata = Step.parse_prep_file(prep_file_path)
+        if 'target_gene' in metadata.columns:
+            tg = set(metadata['sample target_gene'])
+            if len(tg) != 1:
+                raise ValueError("More than one value for target_gene")
+
+            data_type = None
+            for key in Step.AMPLICON_SUB_TYPES:
+                # the key sub_types are often substrings of tg,
+                # rather than equal to tg.
+                if key in tg:
+                    data_type = key
```

Comment:
should this have a break so once found it doesn't continue?

Reply:
I think that's a good idea. However I don't think it's a bad idea to have this sanity check here for a little while as defensive programming.

Reply:
I agree re defensive! If an exact match isnt found, could an exception be raised? I think that would be the most defensive posture

Reply:
From what I've seen, the values in target_gene aren't usually exact matches for {16S, 18S, ITS}, at least for 16S; that's why we search for them as substrings. We do raise an exception if we don't find any matches; do you think we should change to exact match? It's a new column (at least to me) so I'm not sure what to expect yet.

Reply:
Yes, exact match is best. We know the specific target gene, and there any variation in name suggests something was manually filled in or modified upstream, and that leads to noise in Qiita